Symfony/CS/Fixer/Contrib/DirConstantFixer.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+/*
+ * This file is part of the PHP CS utility.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Symfony\CS\Fixer\Contrib;
+
+use Symfony\CS\AbstractFixer;
+use Symfony\CS\Tokenizer\Tokens;
+
+/**
+ * @author Vladimir Reznichenko <kalessil@gmail.com>
+ */
+class DirConstantFixer extends AbstractFixer
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function fix(\SplFileInfo $file, $content)
+    {
+        // generate a new token, otherwise transformations are not possible
+        $dirTokenContainer = Tokens::fromCode('<?php __DIR__; ');
+        $dirConstantToken = $dirTokenContainer->findGivenKind(T_DIR);
+        $dirConstantToken = $dirConstantToken[1];
+
+        $tokens = Tokens::fromCode($content);
+
+        $end = $tokens->count() - 1;
+
+        $sequenceNeeded = array(array(T_STRING, 'dirname'), '(', array(T_FILE), ')');
+
+        $currIndex = 0;
+        while (null !== $currIndex) {
+            $matches = $tokens->findSequence($sequenceNeeded, $currIndex, $end, false);
+
+            // stop looping if didn't find any new matches
+            if (null === $matches) {
+                break;
+            }
+
+            // 0 to 3 accordingly are "dirname", "(", "__FILE__", ")"
+            $matches = array_keys($matches);
+
+            // move cursor just after sequence
+            $currIndex = $matches[3];
+
+            // skip expressions which are not function reference
+            $prevTokenIndex = $tokens->getPrevMeaningfulToken($matches[0]);
+            $prevToken = $tokens[$prevTokenIndex];
+            if ($prevToken->isGivenKind(array(T_DOUBLE_COLON, T_NEW, T_OBJECT_OPERATOR, T_FUNCTION))) {
+                continue;
+            }
+
+            // handle function reference with namespaces
+            if ($prevToken->isGivenKind(array(T_NS_SEPARATOR))) {
+                $twicePrevTokenIndex = $tokens->getPrevMeaningfulToken($prevTokenIndex);
+                $twicePrevToken = $tokens[$twicePrevTokenIndex];
+                if ($twicePrevToken->isGivenKind(array(T_DOUBLE_COLON, T_NEW, T_OBJECT_OPERATOR, T_FUNCTION, T_STRING))) {
+                    continue;
+                }
+            }
+
+            // transform construction
+            $tokens[$matches[0]]->clear();
+            $tokens[$matches[1]]->clear();
+            $tokens[$matches[2]] = $dirConstantToken;
+            $tokens[$matches[3]]->clear();
```

Comment:
@stof : `overrideRange` is not available in 1.9.
I have to re-base onto master (2.0) and target master with this PR.

ping @keradus